### PR TITLE
add ability to set users from oauth token or headers

### PR DESCRIPTION
**Pull request by charleystran**
## Description of change
This change is to allow that if the oauth token is passed and the headers are passed we use the veteran passed in the headers. If the headers are not passed, we use the veteran information from the oauth token.

## Testing done
rspec

## Acceptance Criteria (Definition of Done)
- [x] If the headers are passed along with the oauth token, information for the veteran in the headers are returned
- [x] If only the oauth token is passed, then the veteran information in the Oauth token is returned

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] https://github.com/department-of-veterans-affairs/vets-contrib/issues/1709
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb [hunk 1]
```diff
@@ -47,6 +51,13 @@ def verify_power_of_attorney
           verifier.verify(header('X-Consumer-PoA'))
```
Comment (bastosmichael) [inline modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb]: Should we be getting rid of the PoA header all together?

File: modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb [hunk 2]
```diff
@@ -47,6 +51,13 @@ def verify_power_of_attorney
...
         end
       end
+
+      def use_headers?
+        # if any of the required headers are present we should attempt to use headers
+        request.headers['X-VA-SSN'].present? ||
```
Comment (ATeal) [inline modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb]: you could probably clean this up with an array of headers, and a reduce or something like that

File: modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb (hunk 1 above)
Comment (charleystran) [reply modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb]: I will remove that and change the verifier on this task https://github.com/department-of-veterans-affairs/vets-contrib/issues/1536

File: modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb (hunk 2 above)
Comment (charleystran) [reply modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb]: updated the method

**Pull request merged**